### PR TITLE
Replace `currency` in Network with something more future-proof, e.g. `Symbol`

### PR DESCRIPTION
**Pull request by lszetg**
# Pull Request Template

## Motivation and Context

Explain why this change is necessary or the issue it addresses.

## Summary

Changed ApiNetwork definition according to CIP-0025

## Checklist

Please ensure that your PR meets the following requirements:

### General

- [ ] Code follows project style and guidelines.
- [ ] No redundant or duplicate code.
- [ ] No added new dependencies without clear justification and approval.

### Documentation

- [ ] Added/updated relevant documentation (e.g., README, inline comments).
- [ ] Added a CHANGELOG entry for major changes.

### Testing

- [ ] Added relevant test cases, leveraging tools like `rstest` or similar for reusable tests.
- [ ] Tests pass locally.

### Build & CI/CD

- [ ] Confirmed no CI/CD pipeline issues.
- [ ] Updated build scripts where necessary.

## Comments to Reviewer

Please provide any comments or points of consideration for the reviewers, such as specific areas of focus, potential edge cases, or context that would aid the review process.

### Reviewer Responsibilities

- [ ] Code adheres to style and guidelines.
- [ ] All tests are appropriately covered and they pass.
- [ ] Consider performance and/or security impacts.
- [ ] Constructive feedback is provided.

### CODE REVIEW

File: api_types/src/api/networks.rs [hunk 1]
```diff
@@ -2,19 +2,22 @@ use serde::{Deserialize, Serialize};
...
+    pub display_symbol: String,
+    pub coin_type: u32,
+    pub node_urls: Vec<String>,
+    pub decimals: u32,
+    pub can_do_purchases: bool,
+    pub protocol: ApiProtocol,
     pub block_explorer_url: String,
-    pub enabled: bool,
-    pub network_identifier: Option<String>,
-    pub network_type: ApiNetworkType,
```
Comment (kristifidani) [inline api_types/src/api/networks.rs:19]: Nice 👍  

I would just say that it would be nice to add some doc comments with short explanation on these types and their field 😄

File: sdk/bindings/android/src/type_conversions.rs [hunk 2]
```diff
@@ -11,7 +11,7 @@ pub struct PurchaseDetailsEntity {
     /// Any reasons attached to the status
     pub invalid_reasons: Vec<String>,
     /// The id of the network that the transaction is sent in
-    pub network_id: String,
+    pub network_key: String,
```
Comment (kristifidani) [inline sdk/bindings/android/src/type_conversions.rs:14]: We can update the existing comments  as this is not the id anymore 😄 

`/// The id of the network..`

File: sdk/bindings/swift/src/ffi_functions.rs [hunk 3]
```diff
@@ -89,15 +89,15 @@ impl ETOPaySdk {
     ///
     /// # Arguments
     ///
-    /// * `network_id` - The input string representing the network id.
+    /// * `network_key` - The input string representing the network id.
```
Comment (kristifidani) [inline sdk/bindings/swift/src/ffi_functions.rs:92]: Does the `network_key` represent the `network id`? If not then we need to update the docs.  

This was the case also for the other bindings where we might update the description 😄

File: sdk/src/backend/dlt.rs [hunk 4]
```diff
@@ -163,15 +158,15 @@ mod tests {
...
+            .match_query(Matcher::Exact(format!("network_key={}", iota_network_key)))
             .match_body(Matcher::Exact(body))
             .with_status(status_code)
             .expect(1)
             .with_header("content-type", "application/json")
             .create();
 
         // Act
-        let response = put_user_address(&config, &TOKEN, String::from("67a1f08edf55756bae21e7eb"), ADDRESS).await;
+        let response = put_user_address(&config, &TOKEN, String::from("IOTA"), ADDRESS).await;
```
Comment (kristifidani) [inline sdk/src/backend/dlt.rs:169]: we can use the variable here:  

```suggestion
        let response = put_user_address(&config, &TOKEN, iota_network_key.to_string(), ADDRESS).await;
```

File: sdk/src/core/mod.rs [hunk 5]
```diff
@@ -51,9 +51,9 @@ pub struct Sdk {
     /// Contains the user repository for storing and loading different users.
     repo: Option<UserRepoT>,
     /// The currently active network
-    network: Option<Network>,
+    network: Option<ApiNetwork>,
```
Comment (kristifidani) [inline sdk/src/core/mod.rs:54]: could we rename to `active_network` for better clarity? 😄  

```suggestion
    active_network: Option<ApiNetwork>,
```

File: sdk/src/core/mod.rs [hunk 6]
```diff
@@ -173,8 +172,7 @@ impl Sdk {
 #[cfg(test)]
 mod tests {
     use crate::core::core_testing_utils::handle_error_test_cases;
-    use crate::testing_utils::{example_network_id, example_networks};
-    use crate::types::networks::Network;
+    //use crate::types::networks::Network;
```
Comment (kristifidani) [inline sdk/src/core/mod.rs:175]:
```suggestion
```

File: sdk/src/core/mod.rs [hunk 7]
```diff
@@ -211,10 +210,8 @@ mod tests {
                     username: USERNAME.into(),
                     wallet_manager: Box::new(mock_wallet_manager),
                 });
-                sdk.set_networks(example_networks());
-                sdk.set_network(example_network_id(crate::types::currencies::Currency::Iota))
-                    .await
-                    .unwrap();
+                sdk.set_networks(example_api_networks());
+                sdk.set_network(String::from("IOTA")).await.unwrap();
```
Comment (kristifidani) [inline sdk/src/core/mod.rs:214]: I saw that this `String::from("IOTA")` is hardcoded in a lot of places. I would suggest that we put this in the `testing_utils.rs` and then we get it from there. Then we have only 1 place that is error prune and it would be easier than later to change it. 😄

File: sdk/src/core/viviswap/swap.rs [hunk 8]
```diff
@@ -567,20 +567,21 @@ impl Sdk {
...
     use crate::testing_utils::{
-        example_bank_details, example_contract_response, example_crypto_details, example_exchange_rate_response,
-        example_get_payment_details_response, example_get_user, example_network, example_network_id, example_networks,
+        example_api_network, example_api_networks, example_bank_details, example_contract_response,
+        example_crypto_details, example_exchange_rate_response, example_get_payment_details_response, example_get_user,
         example_viviswap_oder_response, set_config, ADDRESS, AUTH_PROVIDER, HEADER_X_APP_NAME, ORDER_ID, PIN, TOKEN,
         USERNAME,
     };
-    use crate::types::networks::Network;
+    //use crate::types::networks::Network;
```
Comment (kristifidani) [inline sdk/src/core/viviswap/swap.rs:575]:
```suggestion
```

**Review by kristifidani — COMMENTED** [review]
Looks good to me 👍  

I only added some comments which aim better clarity and understanding by mostly adding some doc comments, updating the existing ones. Doing some renaming and avoiding repetition mostly in tests 💯

File: api_types/src/api/dlt.rs [hunk 9]
```diff
@@ -28,14 +28,14 @@ pub struct GetUserAddressResponse {
...
 pub struct SetPreferredNetworkRequest {
     /// The id of the preferred network, or None if it should be cleared.
-    pub network_id: Option<String>,
+    pub network_key: Option<String>,
 }
 
 #[derive(Debug, Deserialize, Serialize, Clone)]
 #[cfg_attr(feature = "utoipa", derive(utoipa::ToSchema))]
 pub struct GetPreferredNetworkResponse {
     /// The id of the preferred network, or None if it is not set.
```
Comment (kristifidani) [inline api_types/src/api/dlt.rs:37]:
```suggestion
    /// The input string representing the network key, or None if it is not set.
```

File: api_types/src/api/dlt.rs [hunk 10]
```diff
@@ -28,14 +28,14 @@ pub struct GetUserAddressResponse {
 #[cfg_attr(feature = "utoipa", derive(utoipa::ToSchema))]
 pub struct SetPreferredNetworkRequest {
     /// The id of the preferred network, or None if it should be cleared.
```
Comment (kristifidani) [inline api_types/src/api/dlt.rs:30]:
```suggestion
    /// The key to the preferred network, or None if it should be cleared.
```

**Review by antbern — APPROVED** [review]

File: sdk/src/types/transactions.rs [hunk 11]
```diff
@@ -56,7 +56,7 @@ pub struct WalletTxInfo {
     pub receiver: String,
     /// Amount of transfer
     pub amount: f64,
-    /// either ETH or IOTA [convert network_id to string based on the value]
+    /// Describes the network
     pub network: String,
```
Comment (kristifidani) [inline sdk/src/types/transactions.rs:60]: FYI this here might need to be `ApiNetwork` type instead of String. But thats another topic which we can do later  😄

**Review by kristifidani — COMMENTED** [review]
Looking much better 👍 Added some small comments which might have slipped along the way 😄

File: sdk/src/types/transactions.rs (hunk 11 above)
Comment (lszetg) [reply sdk/src/types/transactions.rs]: I looked at impl and it should be `network_key` :D it has it's roots in iota transaction :D

**Review by kristifidani — APPROVED** [review]

**Comment by kristifidani** [comment]
Lets merge 🚀

**Review by hauju — APPROVED** [review]
